### PR TITLE
Add async support for subscription

### LINKED ISSUE #549: Support async for subscriptions
see #278

### PR DESCRIPTION
**Pull request by haerdib**
closes #549

### CODE REVIEW

File: examples/examples/event_error_details.rs [hunk 1]
```diff
@@ -13,11 +13,9 @@
 	limitations under the License.
 */
 
-use codec::Decode;
```
Comment (haerdib) [inline examples/examples/event_error_details.rs:16]: Removed unused imports

**Review by echevrier — APPROVED** [review]
LGTM

**Comment by haerdib** [comment]
@Niederb  I'll merge for now. But I'd still be happy if you could have a look at it.

**Comment by Niederb** [comment]
> @Niederb I'll merge for now. But I'd still be happy if you could have a look at it.

@haerdib I had a quick look and it looks quite nice. Much more straightforward than I thought. I will try to implement #348 now. During that I will have a closer look.

**Pull request merged**